generators/app/index.js

```diff
@@ -121,43 +123,47 @@ module.exports = yeoman.generators.Base.extend({
 
      nd.files(source, function (err, files) {
       for ( var i = 0; i < files.length; i++ ) {
-        var filename = files[i].replace(/StarterKit/g, projectName)
+        var ignoreFiles = [];
+		var filename = files[i].replace(/StarterKit/g, projectName)
                                .replace(/starterkit/g, lowerProjectName)
                                .replace('.npmignore', '.gitignore')
                                .replace('dataaccess.ms.json', 'dataaccess.json')
                                .replace('dataaccess.npg.json', 'dataaccess.json')
+							   .replace('DataAccessSettings.ms.cs', 'DataAccessSettings.cs')
+							   .replace('DataAccessSettings.npg.cs', 'DataAccessSettings.cs')
                                .replace(source, dest);
         switch (dataProvider.input)
         {
           case 'p':
-            if (files[i].indexOf('dataaccess.ms.json') > -1 ||
-                files[i].indexOf('dataaccess.ms.json.dist') > -1 ) {
-              console.log(files[i] + ' not created');
-            } else {
-              fs.copy(files[i], filename, copyOptions);
-            }
+            if (files[i].indexOf('dataaccess.ms.json') > -1) {
```

Comment:
Deze constructie kan volgens mij nog simpeler door de ignoreFiles array op te bouwen buiten de for loop. Bv. nd.files vervangen door:

(NIET GETEST dus zal vermoedelijk niet van de eerste keer werken)
```js
    nd.files(source, function (err, files) {
      var ignoreFiles = [];

      switch (dataProvider.input) {
        case 'p':
          ignoreFiles.push(
            'dataaccess.ms.json',
            'DataAccessSettings.ms.cs'
          );
          break;
        case 'm':
          ignoreFiles.push(
            'dataaccess.npg.json',
            'DataAccessSettings.npg.cs'
          );
          break;
        default:
          ignoreFiles.push(
            'EntityContext.cs',
            'DataAccessDefaults.cs',
            'dataaccess.ms.json',
            'dataaccess.npg.json',
            'DataAccessSettings.ms.cs',
            'DataAccessSettings.npg.cs'
          );
      }

      function shouldCopy(file) {
        // Return false if the file is present in the ignoreFiles array
        return !ignoreFiles.some(function ignoreFilesContains(ignoreFile) {
          return file.indexOf(ignoreFile) > -1;
        });
      }

      for (var i = 0; i < files.length; i++) {
        var filename = files[i].replace(/StarterKit/g, projectName)
          .replace(/starterkit/g, lowerProjectName)
          .replace('.npmignore', '.gitignore')
          .replace('dataaccess.ms.json', 'dataaccess.json')
          .replace('dataaccess.npg.json', 'dataaccess.json')
          .replace('DataAccessSettings.ms.cs', 'DataAccessSettings.cs')
          .replace('DataAccessSettings.npg.cs', 'DataAccessSettings.cs')
          .replace(source, dest);

        if (shouldCopy(files[i])) {
          fs.copy(files[i], filename, copyOptions);
        }
      }
    });
```

Als je dit niet simpeler/leesbaarder vindt, zeg het gerust :-)

Reply:
Wat winnen we hiermee ? Performantie ? De output zal in ieder geval dezelfde zijn en dat is het doel van de generators.